File: examples/io_grpc/bazel_build_java/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/dagger/ApplicationGraphModule.java
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.dagger;

import dagger.Binds;
import dagger.Module;

@Module
public interface ApplicationGraphModule {
  @Binds
  GrpcCallScopeGraph.Supplier callGraphSupplier(ApplicationGraph impl);
}

File: examples/io_grpc/bazel_build_kt/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/dagger/ApplicationGraphModule.kt
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.dagger

import dagger.Module
import dagger.Provides

@Module
object ApplicationGraphModule {
  @Provides fun callGraphSupplier(impl: ApplicationGraph): GrpcCallScopeGraph.Supplier = impl
}
